### PR TITLE
[TR] PIM-4321: Resolve the problems of the models override

### PR DESCRIPTION
**Pull request by jjanvier**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        | -
| DB schema updated?   | -
| Migration script?    | -
| Doc PR               | will be done by another card

### CODE REVIEW

File: src/Akeneo/Bundle/StorageUtilsBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -29,6 +29,24 @@ public function getConfigTreeBuilder()
                     ->defaultValue(AkeneoStorageUtilsExtension::DOCTRINE_ORM)
                 ->end()
             ->end()
+            ->children()
+                ->arrayNode('mapping_overrides')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('original')->isRequired()->cannotBeEmpty()
+//                                ->validate()
```
Comment (jjanvier) [inline src/Akeneo/Bundle/StorageUtilsBundle/DependencyInjection/Configuration.php]: fix it

File: src/Akeneo/Bundle/StorageUtilsBundle/EventSubscriber/ConfigureOrmMappingsSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * @author    Julien Janvier <jjanvier@gmail.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * This class is largely based on the Sylius project's class
+ * Sylius/Bundle/ResourceBundle/EventListener/LoadORMMetadataSubscriber.php
+ *
+ * Original authors are Ivan Molchanov <ivan.molchanov@opensoftdev.ru> and Paweł Jędrzejewski <pjedrzejewski@sylius.pl>
+ *
+ * TODO: spec it
```
Comment (jjanvier) [inline src/Akeneo/Bundle/StorageUtilsBundle/EventSubscriber/ConfigureOrmMappingsSubscriber.php]: spec it

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/MongoDBODM/MappingsOverrideConfigurator.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Akeneo\Bundle\StorageUtilsBundle\Doctrine\MongoDBODM;
```
Comment (nidup) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/MongoDBODM/MappingsOverrideConfigurator.php]: directly in Akeneo\Bundle\StorageUtilsBundle\Doctrine\MongoDBODM or subfolder ? (we already have Collections, Cursor, Types) I always have the CatalogBundle Doctrine folder in mind where version after version, we had bunch of classes without same concern at all directly in the folder (maybe my old habits about organizing in lot of subfolder too ;) )

**Comment by nidup** [comment]
GTM

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/MappingsOverrideConfiguratorInterface.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * All the associations of Foo\Bar\Model will be removed and injected in the metadata of Acme\Bar\Model.
+ *
+ * @author    Julien Janvier <jjanvier@gmail.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface MappingsOverrideConfiguratorInterface
+{
+    /**
+     * Configure the mappings of models that are override.
```
Comment (phaseinducer) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/MappingsOverrideConfiguratorInterface.php]: "of overriding models" maybe ?

**Pull request merged**